mitodl/unified-ecommerce
# Add webhook event for add-to-basket

### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/6166


### Description (What does it do?)

Adds a hook implementation that sends some relevant data to the integrated system when the user adds an item to the basket. This is marked as try-last so it should fire _after_ the blocked country check happens.

### How can this be tested?

Automated tests should pass.

Set up a way to catch webhook requests and set up an integrated system to use it. Then, add an item to the cart. This should result in the webhook endpoint being pinged with some data.

Also, complete the order and ensure the webhook gets pinged once the sale is completed as well.

### Additional context

This also refactored the webhook dispatching code so it's more reusable. We still have at least one webhook to implement (for refunds) which this refactoring will help.
